.github/workflows/pull-request.yml

```diff
@@ -132,6 +196,7 @@ jobs:
 
       - name: Setup C++ Build Cache
         id: build-cache
+        if: (steps.filter.outputs.any == 'true')
```

Comment:
one simplification we are doing here is just sunning the setup code every time. we could partition this out further but this seemed fine given that it still finishes before the UI workflow and i didnt want to add unnecessary complexity